tests/bluetooth/audio/cap_commander/src/test_broadcast_reception.c

```diff
@@ -0,0 +1,442 @@
+		test_conn_init(&fixture->conns[i]);
+	}
+}
+
+static void *cap_commander_test_broadcast_reception_setup(void)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture;
+
+	fixture = malloc(sizeof(*fixture));
+	zassert_not_null(fixture);
+
+	return fixture;
+}
+
+static void cap_commander_test_broadcast_reception_before(void *f)
+{
+	memset(f, 0, sizeof(struct cap_commander_test_broadcast_reception_fixture));
+	cap_commander_test_broadcast_reception_fixture_init(f);
+}
+
+static void cap_commander_test_broadcast_reception_after(void *f)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+
+	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		mock_bt_conn_disconnected(&fixture->conns[i], BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
+}
+
+static void cap_commander_test_broadcast_reception_teardown(void *f)
+{
+	free(f);
+}
+
+static void init_start_param(void *f)
+{
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+	int err;
+
+	fixture->start_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	fixture->start_param.param = fixture->start_member_params;
+
+	fixture->start_param.count = ARRAY_SIZE(fixture->start_member_params);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->start_subgroups); i++) {
+		fixture->start_subgroups[i].bis_sync = 1 << i;
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
+		/* TODO: add tests where metadata_len > 0 */
+		fixture->start_subgroups[i].metadata_len = 0;
+#else
+		fixture->start_subgroups[i].metadata_len = 0;
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE */
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fixture->start_member_params); i++) {
+		fixture->start_member_params[i].member.member = &fixture->conns[i];
+		bt_addr_le_copy(&fixture->start_member_params[i].addr, BT_ADDR_LE_ANY);
+		fixture->start_member_params[i].adv_sid = SID;
+		fixture->start_member_params[i].pa_interval = ADV_INTERVAL;
+		fixture->start_member_params[i].broadcast_id = BROADCAST_ID;
+		memcpy(fixture->start_member_params[i].subgroups, &fixture->start_subgroups[0],
+		       sizeof(struct bt_bap_bass_subgroup) * NR_SUBGROUPS);
+		fixture->start_member_params[i].num_subgroups = NR_SUBGROUPS;
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fixture->start_member_params); i++) {
+		fixture->start_member_params[i].member.member = &fixture->conns[i];
+	}
+}
+
+ZTEST_SUITE(cap_commander_test_broadcast_reception, NULL,
+	    cap_commander_test_broadcast_reception_setup,
+	    cap_commander_test_broadcast_reception_before,
+	    cap_commander_test_broadcast_reception_after,
+	    cap_commander_test_broadcast_reception_teardown);
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_start)
+{
+	int err;
+
+	init_start_param(fixture);
```

Comment:
Why not call this in the same way that we call `test_conn_init`?

Reply:
`test_conn_init` only initialises the bt_conn struct, this function initialises the fixture.

I will rename to `test_fixture_init`

Reply:
You may have misunderstood my comment. Instead of calling `init_start_param` in the test and `test_conn_init` in the `cap_commander_test_broadcast_reception_fixture_init`, why not call both in `cap_commander_test_broadcast_reception_fixture_init`?